File: twopointers/26_remove_duplicates_from_sorted_array.java
package twopointers;

// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution {
    public int removeDuplicates(int[] nums) {
        
        if (nums == null) return 0;
        if (nums.length <= 1) return nums.length;

        int left = 0, right = 1, count = 1;
        while (right < nums.length){
            
            if (nums[right] != nums[right-1]){
                nums[++left] = nums[right];
                count++;
            }
            right++;   
        }
        return count;

    }
}

File: twopointers/26_remove_duplicates_from_sorted_array.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/

from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        left, right = 0, 1
        while right < len(nums):
            if (nums[left] != nums[right]):
                left += 1
                nums[left] = nums[right]
            right += 1

        return left + 1